### PR TITLE
Integration of bonells96 Giskard-AI/giskard/PR-291

### PR DESCRIPTION
**Pull request by rabah-khalek**
This PR is the integration of @bonells96's PR: https://github.com/Giskard-AI/giskard/pull/291.
It contains metamorphic tests based on statistical metrics, namely t-test. 

The modifications touch the following files:
- modified:   `giskard/ml_worker/testing/metamorphic_tests.py` (test implementation)
- modified:   `tests/test_metamorphic_invariance.py` (unit test)
- new file:   `giskard/ml_worker/testing/stat_utils.py` (general implementation of t-test metric)

todo-list:
- [x] Divide the stat tests into another submodule instead of merging them with metamorphic? --> these tests, although statistical, they're categorised as metamorphic in the context of Giskard.
- [x] transferring the PR from Giskard to Giskard-client
- [x] Add additional tests for cases of sample sizes < 30:
  - [x] Add another test that is less susceptible to outliers like the average of probability difference.
  - [x] The assumptions of Student's t-test may not be met for small sample sizes. In this case, it is often safer to select a non-parametric test. However, if the assumptions of the t-test are met, it has greater statistical power than Wilcoxon's test.
- [x] Integrate with Giskard’s frontend and check the test with an example.

### CODE REVIEW

**Comment by rabah-khalek** [comment]
- The t-tests have been benchmarked in terms of unit-tests here: https://github.com/Giskard-AI/giskard-client/blob/3137c84719da32a08a25eddc485aa031aacb8ff9/tests/test_metamorphic_invariance.py 
to run the unit test, execute:
```
pytest tests/test_metamorphic_invariance.py -k "test_metamorphic_compare_statistic_tests"
```
- Additional graphical representation is available here: https://github.com/Giskard-AI/giskard-client/blob/graphic_test_output/notebooks/t-tests.ipynb 
Invariance
Decreasing
Increasing

**Review by andreybavt — CHANGES_REQUESTED** [review]
Generally looks good to me, I left a small refactoring comment.
Also we moved the rest of the notebooks to a separate `examples` repo, let's put your new one there too

File: giskard/ml_worker/testing/metamorphic_tests.py [hunk 1]
```diff
@@ -81,6 +83,32 @@ def _compare_prediction(self, results_df, prediction_task, output_sensitivity=No
         failed_idx = results_df.loc[~results_df.index.isin(passed_idx)].index.values
         return passed_idx, failed_idx
 
+    def _compare_probabilities_t_test(self, result_df, flag=None, window_size=0.1, critical_quantile=0.05):
+
+          if flag == 'Invariance':
```
[the remark this answers is not in the document]
Comment (rabah-khalek) [reply giskard/ml_worker/testing/metamorphic_tests.py]: Done.

**Comment by rabah-khalek** [comment]
> Generally looks good to me, I left a small refactoring comment. Also we moved the rest of the notebooks to a separate `examples` repo, let's put your new one there too

This notebook isn't really compatible with examples, it's not a model related one, it's just a graphical visualisation of the test.  I deleted it, and created a new branch https://github.com/Giskard-AI/giskard-client/tree/graphic_test_output in case we want to pursue this later.

**Review by andreybavt — APPROVED** [review]

**Pull request merged**